File: Python/242. Valid Anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        s_dict = {}
        t_dict = {}
        for s_i in s[:]:
            s_dict[s_i]  = s_dict.get(s_i, 0) + 1
        for t_i in t[:]:
            t_dict[t_i]  = t_dict.get(t_i, 0) + 1
        if s_dict == t_dict: return True
        else: return False

File: C++/242. Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map <char, int> s_mp;
        unordered_map <char, int> t_mp;
        for (int i=0; i<s.size(); i++){
            s_mp[s[i]] +=1;
        }
        for (int i=0; i<t.size(); i++){
            t_mp[t[i]] +=1;
        }
        return s_mp==t_mp;
    }
};